index.js

```diff
@@ -223,11 +223,12 @@
           var fn = this.__outsideClickHandler;
           if (typeof document !== 'undefined') {
             var events = this.props.eventTypes || DEFAULT_EVENTS;
+            var passive = !this.props.preventDefault;
```

Comment:
Cant really think of a scenario you have mentioned. If somebody is using composition he should be quite aware of what exactly his compositions are doing and when using it on some shared mutable state (like event) manipulation of the passed state is always possible and should be easily recognizable I think.

Maybe just never had any problem with it. Anyway - gonna update a PR in 2 minutes